system/java_certs.py

```diff
@@ -0,0 +1,234 @@
+    import_cmd = "keytool -importcert -noprompt -keystore '%s' -storepass '%s' -file '%s' -alias '%s'" % (keystore_path, keystore_pass, path, alias)
+
+    # Use local certificate from local path and import it to a java keystore
+    (rc, import_out, import_err) = module.run_command(import_cmd, check_rc=False)
+    if rc == 0:
+        return module.exit_json(changed=True, msg=import_out,
+            rc=rc, cmd=import_cmd, stdout_lines=import_out)
+    else:
+        return module.fail_json(msg=import_out, rc=rc, cmd=import_cmd)
+
+def delete_cert(module, keystore_path, keystore_pass, alias):
+    del_cmd = "keytool -delete -keystore '%s' -storepass '%s' -alias '%s'" % (keystore_path, keystore_pass, alias)
+
+    # Delete SSL certificate from keystore
+    (rc, del_out, del_err) = module.run_command(del_cmd, check_rc=True)
+
+    return module.exit_json(changed=True, msg=del_out,
+        rc=rc, cmd=del_cmd, stdout_lines=del_out)
+
+def test_keytool(module):
+    ''' Test if keytool is actuall executable or not '''
+    test_cmd = "keytool"
+
+    (rc, del_out, del_err) = module.run_command(test_cmd, check_rc=True)
+
+def test_keystore(module, keystore_path):
+    ''' Check if we can access keystore as file or not '''
+    if keystore_path is None:
+        keystore_path=''
+
+    if not os.path.exists(keystore_path) and not os.path.isfile(keystore_path):
+        ## Keystore doesn't exist we want to create it
+        return module.fail_json(changed=False, msg="Module require existing keystore at keystore_path '%s'" % keystore_path)
+
+def main():
+    argument_spec = dict(
+            cert_url = dict(required=False, type='str'),
+            cert_path = dict(required=False, type='str'),
+            cert_alias = dict(required=False, type='str'),
+            cert_port = dict(required=False, default='443'),
+            keystore_path = dict(required=False, type='str'),
+            keystore_pass = dict(required=False, default='changeit', type='str'),
+            keystore_create = dict(required=False, default=False, type='bool'),
+            state = dict(required=False, default='present', choices=['present', 'absent'])
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        mutually_exclusive = [
+                                ['cert_url', 'cert_path']
+                             ],
+    )
+
+    filter = {}
+    url = module.params.get('cert_url')
+    path = module.params.get('cert_path')
+    port = module.params.get('cert_port')
+    cert_alias = module.params.get('cert_alias') or url
+
+    keystore_path = module.params.get('keystore_path')
+    keystore_pass = module.params.get('keystore_pass')
+    keystore_create = module.params.get('keystore_create')
+    state = module.params.get('state')
+
+    if path and not cert_alias:
+      module.fail_json(changed=False, msg="Using local path import from %s requires alias argument."%(keystore_path))
+
```

Comment:
I tried to do this but I'm not sure how to get this working as we could have cert_alias in 2 options.

1) cert_path is specified
    - We require cert_alias
2) cert_url is specified
    - cert_alias is not required as cert_url will be used instead.

So I would need conditional _require_together_ . I have not found way how this could be done.